test/test_circuitzoo.jl

```diff
@@ -0,0 +1,65 @@
+using Test
+using QuantumSavory
+using QuantumSavory.CircuitZoo: EntanglementSwap, Purify2to1, Purify3to1, PurifyStringent, PurifyExpedient
+
+
+const T2=0
+const perfect_pair = (Z1⊗Z1 + Z2⊗Z2) / sqrt(2)
+const perfect_pair_dm = SProjector(perfect_pair)
+const mixed_dm = MixedState(perfect_pair_dm)
+noisy_pair_func(F) = F*perfect_pair_dm + (1-F)*mixed_dm
+
+function entangled_registers(size, purity, T2=0)
+    regA = Register([Qubit() for  _ in 1:size],
+                    [QuantumOpticsRepr() for  _ in 1:size], 
+                    [T2Dephasing(T2) for  _ in 1:size])
+    regB = Register([Qubit() for  _ in 1:size],
+                    [QuantumOpticsRepr() for  _ in 1:size], 
+                    [T2Dephasing(T2) for  _ in 1:size])
+
+    for i in 1:size
+        initialize!((regA[i],regB[i]), noisy_pair_func(purity))
+    end
+    regA, regB
+end
+
+@testset "Keep Pure I Entanglement" begin
+    @testset "Purify2to1" begin
+        for _ in 1:10
+            regA, regB = entangled_registers(2, 1)
+            purificationcircuit = Purify2to1(:X) # original single selection
+            success = purificationcircuit(regA[1], regB[1], regA[2], regB[2])
+            @test success == true
```

Comment:
could you check that the actual state in the registers is the correct purified state, not only that `success==true`

Reply:
that type of more rigorous tests will require learning a bit more about these purification circuits.

Consider the 2to1 circuits. For them, the test is explicitly checking that a given error type is detected.

Same for 3to1 circuits, but they have more types of errors that can be detected. Building upon this notebook might help https://pluto.krastanov.org/basic-purification-circuits.html